test/common/http/http1/codec_impl_test.cc

```diff
@@ -1537,6 +1537,88 @@ TEST_F(Http1ServerConnectionImplTest, UpgradeRequestWithNoBody) {
   codec_->dispatch(buffer);
 }
 
+TEST_F(Http1ServerConnectionImplTest, ConnectRequestNoContentLength) {
+  initialize();
+
+  InSequence sequence;
+  NiceMock<MockRequestDecoder> decoder;
+  EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
+
+  TestHeaderMapImpl expected_headers{
+      {":authority", "host:80"},
+      {":method", "CONNECT"},
+  };
+  EXPECT_CALL(decoder, decodeHeaders_(HeaderMapEqual(&expected_headers), false));
+  Buffer::OwnedImpl buffer("CONNECT host:80 HTTP/1.1\r\n\r\n");
+  codec_->dispatch(buffer);
+
+  Buffer::OwnedImpl expected_data("abcd");
+  Buffer::OwnedImpl connect_payload("abcd");
+  EXPECT_CALL(decoder, decodeData(BufferEqual(&expected_data), false));
+  codec_->dispatch(connect_payload);
+}
+
+// We use the absolute URL parsing code for CONNECT requests, but it does not
+// actually allow absolute URLs.
+TEST_F(Http1ServerConnectionImplTest, ConnectRequestAbsoluteURLNotallowed) {
+  initialize();
+
+  InSequence sequence;
+  NiceMock<MockRequestDecoder> decoder;
+  EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
+
+  Buffer::OwnedImpl buffer("CONNECT http://host:80 HTTP/1.1\r\n\r\n");
+  EXPECT_THROW(codec_->dispatch(buffer), CodecProtocolException);
+}
+
+TEST_F(Http1ServerConnectionImplTest, ConnectRequestWithEarlyData) {
+  initialize();
+
+  InSequence sequence;
+  NiceMock<MockRequestDecoder> decoder;
+  EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
+
+  Buffer::OwnedImpl expected_data("abcd");
+  EXPECT_CALL(decoder, decodeHeaders_(_, false));
+  EXPECT_CALL(decoder, decodeData(BufferEqual(&expected_data), false));
+  Buffer::OwnedImpl buffer("CONNECT host:80 HTTP/1.1\r\n\r\nabcd");
+  codec_->dispatch(buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, ConnectRequestWithTEChunked) {
+  initialize();
+
+  InSequence sequence;
+  NiceMock<MockRequestDecoder> decoder;
+  EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
+
+  // Connect with body is technically illegal, but Envoy does not inspect the
+  // body to see if there is a non-zero byte chunk. It will instead pass it
```

Comment:
Should we eventually block this? TODO? WDYT?

Reply:
We'd have to basically peek at connect payload and reject if it's not an empty chunk.  I'm happy to TODO if you think it's worth it

Reply:
I'm fine either way. I don't know enough about the RFC here to have an opinion. cc @PiotrSikora